contracts/liquidator/Liquidator.sol

```diff
@@ -28,114 +35,212 @@ contract Liquidator is IUniswapV3FlashCallback, PeripheryImmutableState, Periphe
     ) external override {
+        // Verify uniswap callback, recommended security measure
         FlashCallbackData memory decoded = abi.decode(data, (FlashCallbackData));
         CallbackValidation.verifyCallback(factory, decoded.poolKey);
 
         address[] memory assets = decoded.assets;
 
+        // Allow Comet protocol to withdraw USDC(base token) for collateral purchase
+        TransferHelper.safeApprove(comet.baseToken(), address(comet), decoded.amount);
+
         uint256 totalAmountOut = 0;
         for (uint i = 0; i < assets.length; i++) {
             address asset = assets[i];
             uint256 baseAmount = decoded.baseAmounts[i];
 
             if (baseAmount == 0) continue;
 
-            // XXX approve everything all at once?
-            TransferHelper.safeApprove(comet.baseToken(), address(comet), baseAmount);
-
-            uint24 poolFee = getPoolFee(asset);
-
-            // XXX Replace 0 with more meaningful value here
             // XXX if buyCollateral returns collateral amount after change in Comet, no need to check balance
             comet.buyCollateral(asset, 0, baseAmount, address(this));
-            uint256 collateralAmount = ERC20(asset).balanceOf(address(this));
-
-            TransferHelper.safeApprove(asset, address(swapRouter), collateralAmount);
-
-            uint256 amountOut =
-                swapRouter.exactInputSingle(
-                    ISwapRouter.ExactInputSingleParams({
-                        tokenIn: asset,
-                        tokenOut: comet.baseToken(),
-                        fee: poolFee,
-                        recipient: address(this),
-                        deadline: block.timestamp,
-                        amountIn: collateralAmount,
-                        // XXX is baseAmount a good value to pass here?
-                        amountOutMinimum: baseAmount,
-                        sqrtPriceLimitX96: 0
-                    })
-                );
+            uint256 amountOut = swapCollateral(asset);
             totalAmountOut += amountOut;
         }
 
-        uint256 fee = decoded.reversedPair? fee0 : fee1;
+        // We borrow only 1 asset, so one of fees will be 0
+        uint256 fee = fee0 + fee1;
+        // Payback flashloan to Uniswap pool and profit to the caller
         payback(decoded.amount, fee, comet.baseToken(), totalAmountOut, decoded.payer);
     }
 
+    /**
+     * @dev Returns loan to Uniswap pool and sends USDC (base token) profit to caller
+     * @param amount The loan amount that need to be repaid
+     * @param fee The fee for taking the loan
+     * @param token The base token whch was borrowed for successful liquidation
+     * @param amountOut The total amount of base token received after liquidation
+     * @param payer The caller address of liquidation bot
+     */
     function payback(
         uint256 amount,
         uint256 fee,
         address token,
         uint256 amountOut,
         address payer
```

Comment:
Would it be more accurate to call this `recipient`, since it's the receiving address of the funds (paid by `address(this)`)?